src/build/mdn-comments.ts

```diff
@@ -47,66 +45,124 @@ function extractSummary(markdown: string): string {
-      .map((entry) => new URL(entry.name + "/", dirPath));
-  } catch (error) {
-    console.error("Error reading directories:", error);
-    return [];
+async function walkDirectory(dir: URL): Promise<URL[]> {
+  const entries = await fs.readdir(dir, { withFileTypes: true });
+  const parentDirName = dir.pathname.split("/").filter(Boolean).slice(-1)[0];
+  let results: URL[] = [];
+
+  for (const entry of entries) {
+    if (entry.isDirectory()) {
+      if (entry.name === parentDirName) continue;
+      const subDir = new URL(`${entry.name}/`, dir);
+      results = results.concat(await walkDirectory(subDir));
+    } else if (entry.isFile() && entry.name === "index.md") {
+      results.push(new URL(entry.name, dir));
+    }
   }
+
+  return results;
 }
 
-async function getIndexMdContents(
-  folders: URL[],
-): Promise<{ [key: string]: string }> {
-  const results: { [key: string]: string } = {};
+const types: Record<string, string> = {
+  property: "properties",
+  method: "methods",
+  "landing-page": "ignore",
+  interface: "interfaces",
+  overview: "ignore",
+  event: "events",
+};
 
-  for (const folder of folders) {
-    const indexPath = new URL("index.md", folder);
+function generateTypes(content: string): string[] | undefined {
+  const pageType = content.match(/page-type:\s*["']?([^"'\n]+)["']?/)!;
+  const type = pageType[1].split("-").pop()!;
+  const plural = types[type];
+  if (!plural) return;
 
-    try {
-      const content = await fs.readFile(indexPath, "utf-8");
+  return [plural, type];
+}
 
-      // Improved title extraction
-      const titleMatch = content.match(/title:\s*["']?([^"'\n]+)["']?/);
-      const filename = basename(folder.toString());
-      const title = titleMatch
-        ? titleMatch[1].replace(/ extension$/, "")
-        : filename || "";
+function generateSlug(content: string): string[] {
+  const match = content.match(/slug:\s*["']?([^"'\n]+)["']?/)!;
+  const url = match[1].split(":").pop()!;
+  const parts = url.split("/").slice(2); // skip `/en-US/web/api/...`
+  return parts; // Keep only top-level and method
+}
 
-      const summary = extractSummary(content);
-      results[title] = summary;
-    } catch (error) {
-      console.warn(`Skipping ${indexPath}: ${error}`);
-    }
+function ensureLeaf(obj: Record<string, any>, keys: string[]) {
+  let leaf = obj;
+  for (const key of keys) {
+    leaf[key] ??= {};
+    leaf = leaf[key];
   }
+  return leaf;
+}
 
-  return results;
+function insertComment(
+  root: Record<string, any>,
+  slug: string[],
+  summary: string,
+  types?: string[],
+) {
+  if (slug.length === 1 || !types) {
+    const iface = ensureLeaf(root, slug);
+    iface.__comment = summary;
+  } else {
+    const [ifaceName, memberName] = slug;
+    const iface = ensureLeaf(root, [ifaceName, ...types]);
+
+    const mainComment = root[ifaceName]?.__comment;
+    if (mainComment !== summary) {
+      iface[memberName] = { comment: summary };
+    }
+  }
```

Comment:
```suggestion
  paths: string[],
) {
  if (!paths.length) {
    const iface = ensureLeaf(root, slug);
    iface.comment = summary;
  } else {
    const [ifaceName, memberName] = slug;
    const target = ensureLeaf(root, [ifaceName, ...paths, memberName]);
    target.comment = summary;
  }
```

1. `comment` should be used instead of `__comment`.
2. `mainComment !== summary` check doesn't seem to do anything meaningful, it can go.